observations/forward_operators/obs_def_abs_humidity_mod.f90

```diff
@@ -0,0 +1,186 @@
+integer,             intent(out)    :: istatus(ens_size)
+
+real(r8), dimension(ens_size) :: qvap, tmpk, pres, rho_dry, tmpv
+integer,  dimension(ens_size) :: qvap_istatus, tmpk_istatus, pres_istatus
+real(r8)                      :: xyz(3)
+logical :: return_now
+
+if ( .not. module_initialized ) call initialize_module
+
+istatus = 0
+
+!  interpolate the mixing ratio to the location
+call interpolate(state_handle, ens_size, location, QTY_VAPOR_MIXING_RATIO, qvap, qvap_istatus)
+call track_status(ens_size, qvap_istatus, ah, istatus, return_now)
+
+where (istatus == 0 .and. qvap < 0.0_r8) qvap = epsilon(0.0_r8)  ! clamping?
+where (istatus /= 0) istatus = 99
+if(return_now) return
+
+!  interpolate the temperature to the desired location
+call interpolate(state_handle, ens_size, location, QTY_TEMPERATURE, tmpk, tmpk_istatus)
+call track_status(ens_size, tmpk_istatus, ah, istatus, return_now)
+
+where (tmpk <= 0.0_r8)
+   ah = missing_r8
+   istatus = 99
+endwhere
+
+if(all(istatus /= 0)) return  ! not using return_now because where clause modifies istatus.
+
+!  interpolate the pressure, if observation location is not pressure
+if ( is_vertical(location, "PRESSURE") ) then
+   xyz = get_location(location)
+   pres = xyz(3)
+else
+   ! pressure comes back in pascals (not hPa or mb)
+   call interpolate(state_handle, ens_size, location, QTY_PRESSURE, pres, pres_istatus)
+   call track_status(ens_size, pres_istatus, ah, istatus, return_now)
+
+   where (pres <= 0.0_r8 .or. pres >= 120000.0_r8)
+      ah = missing_r8
+      istatus = 99
+   end where
+   if(all(istatus /= 0)) return  ! not using return_now because where clause modifies istatus.
+
+endif
+
+!  Compute the ah
+where (istatus == 0)
+   tmpv = tmpk * (1.0_r8 + qvap * gas_constant_v / gas_constant)
+   rho_dry = pres / (gas_constant * tmpv)
+   ah = qvap * rho_dry
+end where
+
+! Warnings - first time only
+
+! nsc - my dilemma: each task does these obs independently.  if you use E_ALLMSG you'll
+! get a message from every processor which runs into this case.  if you use E_MSG you'll
+! only see those on task 0.  for now i've removed the test and you'll get the message (once)
+! even if there aren't any values outside the valid range.  i can't see a good way to
+! enforce that you get only one message when this could happen on any task and probably
+! will happen on many tasks.
+
+if (first_time_warn_low) then
+   !if (any(ah < MIN_VALUE .and. istatus == 0)) then
+   !   write(msgstring, '(A,F12.6)') 'values lower than low limit detected, e.g.', minval(ah, istatus==0)
+      write(msgstring, *) 'checking absolute humidity value computed by the forward operator'
+      call error_handler(E_MSG,'get_expected_absolute_humidity', msgstring,      &
+                         text2='all values lower than 0 will be set to 0', &
+                         text3='this message will only print once')
+      first_time_warn_low = .false.
+   !endif
```

Comment:
should the if statement be there or not ... commenting out the stuff is not in keeping with our coding practice.

Reply:
also inherited from obs_def_rel_humidity_mod.f90. I turned back on the if statement.